### PR TITLE
policy: add documentation to L4Filter type

### PR DESCRIPTION
**Pull request by ianvernon**
No functional changes.

Signed-off by: Ian Vernon <ian@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8466)

### CODE REVIEW

**Review by joestringer — COMMENTED** [review]
L4Filter is also used to represent all ports/protocols for allow-all, right? The "specific L4 port/protocol" suggests it only applies to one port/protocol, but not the wildcarding cases.

**Comment by ianvernon** [comment]
@joestringer noted that it can allow at all ports and protocols as well.

File: pkg/policy/l4.go [hunk 1]
```diff
@@ -95,8 +95,14 @@ const (
 	ParserTypeDNS L7ParserType = "dns"
 )
 
+// L4Filter represents the policy (allowed remote sources / destinations of
+// traffic) that applies at a specific L4 port/protocol combination (including
+// all ports and protocols), at either ingress or egress. The policy here is
+// still specified in terms of selectors (i.e, not in terms of security
+// identities).
```
Comment (jrajahalme) [inline pkg/policy/l4.go]: Could change the end to read:
```
The policy here is specified in terms of selectors that are mapped
to security identities via the selector cache.
 ```

**Review by jrajahalme — APPROVED** [review]
Small improvement proposal below.

**Review by joestringer — APPROVED** [review]
I like Jarno's suggestion too. But LGTM.

File: pkg/policy/l4.go (hunk 1 above)
Comment (ianvernon) [reply pkg/policy/l4.go]: Done!

**Pull request merged**